metalnetlet/controllers/networkinterface_controller.go

```diff
@@ -161,6 +166,130 @@ func (r *NetworkInterfaceReconciler) getLoadBalancerTargetsForNetworkInterface(c
 	return ips, nil
 }
 
+func (r *NetworkInterfaceReconciler) getNetworkPolicyRulesForNetworkInterface(ctx context.Context, nic *v1alpha1.NetworkInterface) ([]metalnetv1alpha1.FirewallRule, error) {
+	var firewallRules []metalnetv1alpha1.FirewallRule
+
+	npRuleList := &v1alpha1.NetworkPolicyRuleList{}
+	if err := r.List(ctx, npRuleList,
+		client.InNamespace(nic.Namespace),
+	); err != nil {
+		return nil, fmt.Errorf("error listing network policy rules: %w", err)
+	}
+
+	for _, npRule := range npRuleList.Items {
+		hasDst := slices.ContainsFunc(npRule.Targets,
+			func(target v1alpha1.TargetNetworkInterface) bool {
+				return slices.Contains(nic.Spec.IPs, target.IP)
+			},
+		)
+		if hasDst {
+			rules := getFirewallRulesFromNetworkPolicyRule(&npRule)
+			firewallRules = append(firewallRules, rules...)
+		}
+	}
+
+	return firewallRules, nil
+}
+
+func getFirewallRulesFromNetworkPolicyRule(npRule *v1alpha1.NetworkPolicyRule) []metalnetv1alpha1.FirewallRule {
+	var firewallRules []metalnetv1alpha1.FirewallRule
+	priority := npRule.Priority
+
+	for _, ingressRule := range npRule.IngressRules {
+		rules := extractFirewallRulesFromRule(ingressRule, metalnetv1alpha1.FirewallRuleDirectionIngress, priority)
+		firewallRules = append(firewallRules, rules...)
+	}
+
+	for _, egressRule := range npRule.EgressRules {
+		rules := extractFirewallRulesFromRule(egressRule, metalnetv1alpha1.FirewallRuleDirectionEgress, priority)
+		firewallRules = append(firewallRules, rules...)
+	}
+
+	return firewallRules
+}
+
+func extractFirewallRulesFromRule(rule v1alpha1.Rule, direction metalnetv1alpha1.FirewallRuleDirection, priority *int32) []metalnetv1alpha1.FirewallRule {
+	var firewallRules []metalnetv1alpha1.FirewallRule
+
+	for _, port := range rule.NetworkPolicyPorts {
+		firewallRule := metalnetv1alpha1.FirewallRule{
+			FirewallRuleID: types.UID(uuid.New().String()),
+			Direction:      direction,
+			Action:         metalnetv1alpha1.FirewallRuleActionAccept,
+			Priority:       priority,
+			IpFamily:       corev1.IPv4Protocol, //TODO: later support for IPv6
```

Comment:
Can't we derive the `IPFamily` from the `ironcore-net` `Rule` object instead of pinning it to IPv4?

Reply:
Done, derived IPFamily from Prefix.